### ISSUE TITLE
Enable handling of unicode characters

### ISSUE THREAD

**Issue by bewedeg** [tonyfarney/FixedLengthFileHandler#1]
Hi Tony, 
I think i might have found a small issue in 
https://github.com/tonyfarney/FixedLengthFileHandler/blob/3c999d3bc7ef6373f018abbfc523d59e7f4cc64b/src/FixedLengthFileReader.php#L62
In Germany and other countries there are special characters like ä, ö, ü, ß and so on. If the underlying file has these encoded in unicode `substr ` in PHP does not handle these characters very nicely, as the character bytewise is cut in half leaving garbage. PHP has [mb_substr(...)](https://www.php.net/manual/de/function.mb-substr.php) to handle this character in the correct manner.

I replaced that in a local copy and i can confirm that it works as expected.
`				$fieldConfig, mb_substr($line, $i, $fieldConfig['size'])
`

Regards,
Bernhard

**Comment by tonyfarney** [comment]
You are right! The library is not ready to work with multi byte encodings. Please feel free to issue a PR.

**Comment by tonyfarney** [comment]
You are right! The library lacks in providing support for that kind of situation. I am not used with those characters, but it appears to be an encoding issue. In the case you pointed out, my understanding is that a symbol that is considered to be 1 character is actually encoded as 2 characters. Is that the case? If so, I think that the correct approach is to use an encoding that properly represents that simbols as 1 character.

**Comment by bewedeg** [comment]
I just looked at the file and its plain ANSI 
![image](https://user-images.githubusercontent.com/2355906/171236769-547ed777-f089-4d1d-8134-165830dd2b5b.png)
**Ü** is stored as HEX **dc**

which leaves me even more puzzled why `substr` doesn't handle this case correct but `mb_substr` does.

**Comment by tonyfarney** [comment]
Interesting. If just replacing by mb_substr solved the issue, please feel free to issue a PR. But it's necessary to check if the function exists (as it is part of the mbstring extension) before using it. If it's not defined, fallback to substr.
